site/src/pages/components/beforefocus.explainer.mdx

```diff
@@ -0,0 +1,137 @@
+
+While having additional new high level abstractions to handle keyboard navigation (like,
+for example, [focusgroup](./focusgroup.explainer)) sounds plausible, in reality it
+would be hard or impossible to align on the APIs that would cover all necessary cases —
+there is a multitude of different applications and no single standardized approach for them.
+It would also involve a huge body of work in the browser engine.
+
+Instead of high level APIs, this proposal focuses on having two fairly simple lower level
+functions which would arguably allow the application developers to implement whatever patterns
+they prefer without working around the browser limitations and impossibilities.
+
+Namely, one method called `document.focusNext()` and one event called `beforefocus`.
+Let's dive deeper into it.
+
+## document.focusNext()
+
+`document.focusNext()` method just exposes the default action the browsers have for `Tab` and
+`Shift+Tab` presses.
+
+The method's signature should be like:
+
+```ts
+focusNext(options?: {
+   contextElement?: HTMLElement,
+   direction?: FORWARD | BACKWARD,
+   trapped?: boolean
+});
+```
+
+`contextElement` is the starting point from which the browser should start looking for the next
+element to focus. It is optional and equals to `document.activeElement` by default.
+
+`direction` is the direction of the search. `FORWARD` to get the `Tab` press behaviour and
+`BACKWARD` to get `Shift+Tab` press behaviour. It is also optional and equals to `FORWARD` by
+default.
+
+`trapped` means that instead of going to the address bar after the last element, the search
+should continue from the beginnig of the page (or from the end of it in case of moving
+backwards). False by default.
+
+This method should invoke exactly the same internal routine to locate and focus
+elements which the currently existing default action for `Tab` and `Shift+Tab` invokes.
+
+The important expected behaviours:
+
+* If the `contextElement` is the last element in the DOM, calling `document.focusNext()` moves
+   focus outside of the page to the browser's address bar — the same thing `Tab` press default
+   action does.
+* If an `<iframe>` comes after the `contextElement`, calling `document.focusNext()` should
+   move focus to the first focusable element inside the iframe (if the iframe has focusable
+   elements) or past the iframe (if there is nothing to focus in the iframe) — the same thing
+   `Tab` press default action does.
+* The same logic applies for mirroring `Shift+Tab` behaviour when `direction` is set ot `BACKWARD`.
+* `document.focusNext()` should have no effect if the page is not focused or the focus is in
+   the address bar.
+* When `trapped` is true, the focus should cycle inside the page, skipping the address bar.
+
+Implementing `document.focusNext()` will be fairly straightforward as no new behaviour needs
+to be added, just the one that is already present should be reused.
```

Comment:
I don't think this sentence is necessary, I also worry it might read as presumptuous to make a claim about how straightforward an API would be to implement.